test/python/quantum_info/states/test_utils.py

```diff
@@ -55,6 +55,35 @@ def test_shannon_entropy(self):
         # Base 10
         self.assertAlmostEqual(0.533908120973504, shannon_entropy(input_pvec, 10))
 
+    def test_schmidt_decomposition(self):
```

Comment:
Could you test to see if you are getting the correct decomposition, not just the return to the original.

Reply:
I went ahead and implemented the test as described above, but let me know if you would like to see it implemented in a different way. Thanks!

Reply:
Thanks. AFAIK, there are no rules or conventions on that point. I'm fine with this implementation. (I'd like to hear other's opinion.)

Reply:
Hi @ikkoham. I found this very helpful report on the sign ambiguity of the SVD:
https://www.osti.gov/servlets/purl/920802

What they suggest to deal with the sign ambiguity, is: 
_"...in order to identify the sign of a singular vector, it is suggested that it be similar to the sign of the majority of vectors it is representing."_

They propose a function to do this, but the issue is that they only consider real-valued matrices. I am not familiar with how SVD algorithms work, so I don't know if in the case of complex-valued data, the SVD could also return singular vectors with arbitrary phases as long long as the overall sign of the sum term is preserved. If this is the case, the implementation of this function can get complicated.

Reply:
Instead of checking for equality between individual statevectors perhaps it would be better to just verify that the result does indeed recompose and satisfy all properties as described in your documentation of a Schmidt decomposition. 

The question of numerical stability of the decomposition due to the sign issue could be treated in a separate test since it addresses a somewhat distinct issue. I'm not yet familiar with what the best convention might be at the moment so I'd be ok with any convention, which might include unhandled, and clearly documenting the choice.

Also in your current test function it would be better to split it up into several tests along the lines of each commented block you already have.

Reply:
Hi @ewinston. Thanks for the feedback. 

1. Currently there are tests for both: 1) check for equality between individual singular vectors and singular values, **and** 2) test to verify that the result recomposes. Is your suggestion to get rid of 1)?
2. Regarding sign ambiguity testing, do you mean we should just keep the test we have right now but move it under a different test function? Or are you referring to having a different set of tests altogether?
3. Regarding sign convention selection, I am not quite sure what algorithm is used in the numpy implementation of the svd, but in the paper I shared it is mentioned that if it is a Lanczos-based method, there is a random component to it, so the selected sign might not be guaranteed. The best solution would be to implement the sign correction process suggested in that paper, but again, here we're just trying to correct for some "global" phase factor which is less critical in quantum computing compared to other applications where the sign might play an important role.
4. I will split the test functions just as you suggested. I will probably wait to hear back from you regarding items 1 and 2 before doing so, so I can make the final changes all together.